test/e2e/framework/ginkgowrapper.go

```diff
@@ -31,3 +37,114 @@ func AnnotatedLocation(annotation string) types.CodeLocation {
 	codeLocation = types.NewCustomCodeLocation(annotation + " | " + codeLocation.String())
 	return codeLocation
 }
+
+// It is a wrapper around ginkgo.It which supports [WithFeatureGate],
+// [WithFeature], [WithEnvironment], [WithConformance], [WithNodeConformance]
+// as optional arguments in addition to those already supported by ginkgo
+// itself, like [ginkgo.Label].
+func It(text string, args ...interface{}) {
+	var ginkgoArgs []interface{}
+	var offset ginkgo.Offset
+
+	addLabel := func(label string) {
+		text += fmt.Sprintf(" [%s]", label)
+		ginkgoArgs = append(ginkgoArgs, ginkgo.Label(label))
+	}
+
+	for _, arg := range args {
+		switch arg := arg.(type) {
+		case withFeature:
+			addLabel("Feature:" + string(arg))
+		case withFeatureGate:
+			addLabel("FeatureGate:" + string(arg.featureGate))
+			addLabel(arg.level)
+		case withEnvironment:
+			addLabel("Environment:" + string(arg))
+		case withConformance:
+			addLabel("Conformance")
+		case withNodeConformance:
+			addLabel("NodeConformance")
+		case withDisruptive:
+			addLabel("Disruptive")
+		case withSerial:
+			addLabel("Serial")
+			ginkgoArgs = append(ginkgoArgs, ginkgo.Serial)
+		case ginkgo.Offset:
+			offset = arg
+		default:
+			ginkgoArgs = append(ginkgoArgs, arg)
+		}
+	}
+
+	offset++
+	ginkgoArgs = append(ginkgoArgs, offset)
+	ginkgo.It(text, ginkgoArgs...)
+}
+
+// TODO: Describe and Context wrappers.
+
+func WithFeature(name string) withFeature {
+	// TODO: verify that the feature is known.
+	return withFeature(name)
+}
+
+// WithFeatureGate adds a certain feature gate as label to a [framework.It].
+// The feature gate must be listed in
+// [k8s.io/apiserver/pkg/util/feature.DefaultMutableFeatureGate].
+func WithFeatureGate(featureGate featuregate.Feature) withFeatureGate {
+	spec, ok := utilfeature.DefaultMutableFeatureGate.GetAll()[featureGate]
+	if !ok {
+		panic(fmt.Sprintf("invalid WithFeatureGate call: the feature gate %q is unknown", featureGate))
+	}
+
+	// We use mixed case (i.e. Beta instead of BETA) and Stable instead of
+	// the empty string.
+	var level string
+	if spec.PreRelease == "" {
+		level = "Stable"
```

Comment:
my vote is about not tagging stables at all, as they must be part of the core system I don't see the value in the feature gate tags

Reply:
I agree. I added it only because https://github.com/kubernetes/enhancements/pull/3042 has it. The actual usage is not clear to me yet.

Reply:
this is for those rare cases when feature gate is GA, but not locked to default.

Reply:
So a GA feature that can still be disabled? I can see why continuing to list such a feature as test tag may be useful.

But how does marking such a feature as "Stable" help? In other words, how would the "Stable" tag be used? If it's not used, we don't need it...

Reply:
I'm not attached to the tag name. Just thought we need something so we can find jobs that set the `FeatureGate`, but haven't set it's stage

Reply:
My proposal in this PR is to derive the stage from the feature, i.e. it never would be set explicitly.

Reply:
But we still need to clarify what's supposed to happen when a feature graduates to GA and then later gets removed from the set of defined features.

Reply:
>my vote is about not tagging stables at all

Agree, I feel like this could imply `[Stable]` should be added to everything that is currently not `[Alpha]`, `[Beta]` or `[Deprecated]` and I don't see the value in it.  Also `featuregate.GA` is already `""` so we're keeping parity in some sense.

> this is for those rare cases when feature gate is GA, but not locked to default.

Some examples to motivate discussion: https://github.com/kubernetes/kubernetes/blob/5113b705d2b2c96f04f9eca5ad427ba3b441ae23/pkg/features/kube_features.go#L872-L879

> Just thought we need something so we can find jobs that set the FeatureGate, but haven't set it's stage

regex or label equivalent to `+FeatureGate:.* -Alpha -Beta -Deprecated` but again I'm not sure why we would want or need this. 

If we set the convention that everything uses `WithFeatureGate` which auto-derives alpha/beta/deprecated I'm not sure this matters so much?

>we still need to clarify what's supposed to happen when a feature graduates to GA and then later gets removed from the set of defined features

Same thing, no stable/GA label? I think you're talking Feature, not FeatureGate here, and I'm not entirely sure why the difference matters.

Reply:
I was talking FeatureGate and how it transitions (or doesn't?) to Feature.

Let's look at a fictional test that depends on FeatureGate foo. While that is in alpha/beta, we annotate the test with `[FeatureGate: foo] [alpha/beta]`, with '[alpha/beta]' derived from the definition in `kube_features.go`.

Now it graduates to GA and remains in `kube_features.go`, either because it isn't locked to its default yet or because all features are kept around a bit longer to avoid breaking code that uses `--feature-gates foo=true`. During that time we annotate the test with `[FeatureGate: foo]` and nothing else.

The last step is removal from `kube_features.go`. Now we remove `[FeatureGate: foo]`. We have to do something, because it won't compile anymore.

Or do we turn it into `[Feature: foo]`? In all cases or only where it matters? It matters if the feature, even though GA, is not available everywhere.

For features that are available everywhere, adding `[Feature: foo]` has the advantage that it becomes easy to run all tests related to a certain feature. The downside is that many jobs currently exclude tests marked with `[Feature: <something>]` and would therefore exclude the feature. I'm undecided.

Reply:
I think that we should not create this relation Feature <---> FeatureGate, those are independent things that some times be related, a FeatureGate doesn't have to be related to a Feature necessarily.

Reply:
So once a feature gate gets removed from `kube_features.go`, we also remove the test annotation. Works for me.